### PR TITLE
Changelog with shapes

### PR DESCRIPTION
**Pull request by smizell**
This adds support for gathering endpoint changes when there are shape changes.

### CODE REVIEW

File: workspaces/graph-lib/src/endpoints-graph/index.ts [hunk 1]
```diff
@@ -125,6 +125,27 @@ export interface NodeListWrapper {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+export class BodyNodeWrapper implements NodeWrapper {
+  constructor(public result: Node, private queries: GraphQueries) {
+  }
+
+  response(): PathNodeWrapper | null {
```
Comment (devdoshi) [inline workspaces/graph-lib/src/endpoints-graph/index.ts:132]: this should be ResponseNodeWrapper

File: workspaces/graph-lib/src/endpoints-graph/index.ts [hunk 2]
```diff
@@ -125,6 +125,27 @@ export interface NodeListWrapper {
...
+    const neighbors = this.queries.listOutgoingNeighborsByType(this.result.id, NodeType.Response);
+    if (neighbors.results.length === 0) {
+      return null;
+    }
+    return neighbors.results[0] as PathNodeWrapper;
+  }
+
+  request(): PathNodeWrapper | null {
```
Comment (devdoshi) [inline workspaces/graph-lib/src/endpoints-graph/index.ts:140]: This should be RequestNodeWrapper

File: workspaces/graph-lib/src/shapes-graph/index.ts [hunk 3]
```diff
@@ -248,6 +258,8 @@ export class GraphQueries {
       return new ShapeParameterNodeWrapper(node, this);
     } else if (node.type === NodeType.Field) {
       return new FieldNodeWrapper(node, this);
+    }else if (node.type === NodeType.BatchCommit) {
```
Comment (devdoshi) [inline workspaces/graph-lib/src/shapes-graph/index.ts:261]: Please use our .prettierrc

File: workspaces/spectacle/test/index.ts [hunk 4]
```diff
@@ -18,3 +71,40 @@ Tap.test('spectacle batchCommits query', async (test) => {
   });
   test.matchSnapshot(results);
 });
+
+specs.forEach(async (spec) => {
+  Tap.test(`spectacle changelog query ${spec.name}`, async (test) => {
+    const spectacle = await makeSpectacle(OpticEngine, {
```
Comment (devdoshi) [inline workspaces/spectacle/test/index.ts:77]: this probably doesn't build anymore. You might have overwritten it but I replaced the usage in this file with InMemoryOpticContextBuilder

**Review by devdoshi — APPROVED** [review]
just a few small changes

**Comment by smizell** [comment]
Comments addressed!

**Pull request merged**